### PR TITLE
Read AABB Info From Model File(obj/c3t/c3b)

### PR DESCRIPTION
**Pull request by songchengjiang**

### CODE REVIEW

**Comment by lvlonggame** [comment]
Good job!!!

File: cocos/3d/CCBundle3DData.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "base/CCRef.h"
 #include "base/ccTypes.h"
 #include "math/CCMath.h"
+#include "CCAABB.h"
```
Comment (super626) [inline cocos/3d/CCBundle3DData.h]: should be "3d/CCAABB.h"

File: cocos/3d/CCBundle3D.cpp [hunk 2]
```diff
@@ -424,6 +426,21 @@ bool  Bundle3D::loadMeshDatasBinary(MeshDatas& meshdatas)
             std::vector<unsigned short>      indexArray;
             std:: string meshPartid = _binaryReader.readString();
             meshData->subMeshIds.push_back(meshPartid);
+
+            if (_version != "0.3" && _version != "0.4")
+            {
+                unsigned int aabbSize=0;
+                //read mesh aabb
+                if (_binaryReader.read(&aabbSize, 4, 1) != 1 || aabbSize < 6)
```
Comment (super626) [inline cocos/3d/CCBundle3D.cpp]: aabbSize != 6 is better.
In fact, it is not very necessary to store the aabb size.

File: cocos/3d/CCMeshVertexIndexData.cpp [hunk 3]
```diff
@@ -106,12 +106,16 @@ MeshVertexData* MeshVertexData::create(const MeshData& meshdata)
...
     AABB aabb;
+    bool needCalcAABB = meshdata.subMeshAABB.size() == meshdata.subMeshIndices.size()? false: true;
     for (size_t i = 0; i < meshdata.subMeshIndices.size(); i++) {
-        
+
         auto& index = meshdata.subMeshIndices[i];
         auto indexBuffer = IndexBuffer::create(IndexBuffer::IndexType::INDEX_TYPE_SHORT_16, (int)(index.size()));
         indexBuffer->updateIndices(&index[0], (int)index.size(), 0);
-        aabb = MeshVertexData::calculateAABB(meshdata.vertex, meshdata.getPerVertexSize(), index);
+        if (needCalcAABB)
```
Comment (super626) [inline cocos/3d/CCMeshVertexIndexData.cpp]: We can compute the aabb in the bundle if we do not have a aabb in the file.
Then we can just assign it to the MeshIndexData here.

**Comment by songchengjiang** [comment]
fixed

**Pull request merged**